triton-vm/src/proof.rs

```diff
@@ -39,3 +67,107 @@ impl Claim {
         self.output.iter().map(|&x| BFieldElement::new(x)).collect()
     }
 }
+
+impl BFieldCodec for Claim {
+    fn decode(sequence: &[BFieldElement]) -> anyhow::Result<Box<Self>> {
+        if sequence.len() < 7 {
+            bail!(
+                "Cannot decode Vec of {} < 7 BFieldElements as Claim",
+                sequence.len()
+            );
+        }
+        let program_digest = *Digest::decode(&sequence[0..DIGEST_LENGTH])?;
+        let mut read_index = DIGEST_LENGTH;
+
+        let input_length = sequence[read_index].value() as usize;
+        if sequence.len() < read_index + input_length + 1 {
+            bail!("Cannot decode Vec of BFieldElements as Claim: improper input length");
+        }
+        read_index += 1;
+        let input = sequence[read_index..read_index + input_length]
+            .to_vec()
+            .iter()
+            .map(|b| b.value())
+            .collect_vec();
+        read_index += input_length;
+
+        let output_length = sequence[read_index].value() as usize;
+        if sequence.len() < read_index + output_length {
+            bail!("Cannot decode Vec of BFieldElements as Claim: improper output length");
+        }
+        read_index += 1;
+        let output = sequence[read_index..read_index + output_length]
+            .to_vec()
+            .iter()
+            .map(|b| b.value())
+            .collect_vec();
+
+        Ok(Box::new(Claim {
+            program_digest,
+            input,
+            output,
+            padded_height: 0,
+        }))
+    }
+
+    fn encode(&self) -> Vec<BFieldElement> {
+        let mut string = self.program_digest.encode();
+        string.push(BFieldElement::new(self.input.len() as u64));
+        string.append(&mut self.public_input());
+        string.push(BFieldElement::new(self.output.len() as u64));
+        string.append(&mut self.public_output());
+        string
+    }
+}
+
+#[cfg(test)]
+pub mod test_claim_proof {
+    use itertools::Itertools;
+    use rand::random;
+    use twenty_first::shared_math::{
+        b_field_element::BFieldElement, bfield_codec::BFieldCodec, other::random_elements,
+        tip5::Digest,
+    };
```

Comment:
```suggestion
    use twenty_first::shared_math::b_field_element::BFieldElement;
    use twenty_first::shared_math::bfield_codec::BFieldCodec;
    use twenty_first::shared_math::other::random_elements;
    use twenty_first::shared_math::tip5::Digest;
```

For some reason, I find the “use a::{b::{c::d, e::f}}” syntax comparatively difficult to parse.